src/rdkafka_sasl_oauthbearer.c

```diff
@@ -0,0 +1,1006 @@
+        rd_chariov_t *out) {
+        struct rd_kafka_sasl_oauthbearer_state *state =
+                rktrans->rktrans_sasl.state;
+
+        /*
+         * https://tools.ietf.org/html/rfc7628#section-3.1
+         * kvsep          = %x01
+         * key            = 1*(ALPHA)
+         * value          = *(VCHAR / SP / HTAB / CR / LF )
+         * kvpair         = key "=" value kvsep
+         * ;;gs2-header     = See RFC 5801
+         * client-resp    = (gs2-header kvsep *kvpair kvsep) / kvsep
+         */
+
+        static const char *gs2_header = "n,,";
+        static const char *kvsep = "\x01";
+        const int kvsep_size = strlen(kvsep);
+        int extension_size = 0;
+        int i;
+        char *buf;
+        int size_written;
+
+        for (i = 0 ; i < rd_list_cnt(&state->extensions) ; i++) {
+                rd_strtup_t *extension = rd_list_elem(&state->extensions, i);
+                // kvpair         = key "=" value kvsep
+                extension_size += strlen(extension->name) + 1 // "="
+                        + strlen(extension->value) + kvsep_size;
+        }
+
+        // client-resp    = (gs2-header kvsep *kvpair kvsep) / kvsep
+        out->size = strlen(gs2_header) + kvsep_size
+                + strlen("auth=Bearer ") + strlen(state->token_value)
+                + kvsep_size + extension_size + kvsep_size;
+        out->ptr = rd_malloc(out->size+1);
+
+        buf = out->ptr;
+        size_written = 0;
+        size_written += rd_snprintf(buf, out->size+1 - size_written,
+                "%s%sauth=Bearer %s%s", gs2_header, kvsep, state->token_value,
+                kvsep);
+        buf = out->ptr + size_written;
+
+        for (i = 0 ; i < rd_list_cnt(&state->extensions) ; i++) {
+                rd_strtup_t *extension = rd_list_elem(&state->extensions, i);
+                size_written += rd_snprintf(buf, out->size+1 - size_written,
+                        "%s=%s%s", extension->name, extension->value, kvsep);
+                buf = out->ptr + size_written;
+        }
+
+        rd_snprintf(buf, out->size+1 - size_written, "%s", kvsep);
+
+        rd_rkb_dbg(rktrans->rktrans_rkb, SECURITY, "SASLOAUTHBEARER",
+                   "Built client first message");
+}
+
+
+
+/**
+ * @brief SASL OAUTHBEARER client state machine
+ * @returns -1 on failure (errstr set), else 0.
+ */
+static int rd_kafka_sasl_oauthbearer_fsm (rd_kafka_transport_t *rktrans,
+                                    const rd_chariov_t *in,
+                                    char *errstr, size_t errstr_size) {
+        static const char *state_names[] = {
+                "client-first-message",
+                "server-first-message",
+                "server-failure-message",
+        };
+        struct rd_kafka_sasl_oauthbearer_state *state =
+                rktrans->rktrans_sasl.state;
+        rd_chariov_t out = RD_ZERO_INIT;
+        int r = -1;
+
+        rd_rkb_dbg(rktrans->rktrans_rkb, SECURITY, "SASLOAUTHBEARER",
```

Comment:
be consistent with the fac, suggest just using "OAUTHBEARER".

Reply:
Fixed (twice in this file)